core/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -377,154 +372,173 @@ public boolean isClosed() {
-                        }
-                        boolean success = false;
-                        try {
-                            // connect to the node, see if we manage to do it, if not, bail
-                            if (!nodeFoundByAddress) {
-                                logger.trace("[{}] connecting (light) to {}", sendPingsHandler.id(), finalNodeToSend);
-                                transportService.connectToNodeAndHandshake(finalNodeToSend, timeout.getMillis());
-                            } else {
-                                logger.trace("[{}] connecting to {}", sendPingsHandler.id(), finalNodeToSend);
-                                transportService.connectToNode(finalNodeToSend);
-                            }
-                            logger.trace("[{}] connected to {}", sendPingsHandler.id(), node);
-                            if (receivedResponses.containsKey(sendPingsHandler.id())) {
-                                // we are connected and still in progress, send the ping request
-                                sendPingRequestToNode(sendPingsHandler.id(), timeout, pingRequest, latch, node, finalNodeToSend);
-                            } else {
-                                // connect took too long, just log it and bail
-                                latch.countDown();
-                                logger.trace("[{}] connect to {} was too long outside of ping window, bailing",
-                                        sendPingsHandler.id(), node);
-                            }
-                            success = true;
-                        } catch (ConnectTransportException e) {
-                            // can't connect to the node - this is a more common path!
-                            logger.trace(
-                                (Supplier<?>) () -> new ParameterizedMessage(
-                                    "[{}] failed to connect to {}", sendPingsHandler.id(), finalNodeToSend), e);
-                        } catch (RemoteTransportException e) {
-                            // something went wrong on the other side
-                            logger.debug(
-                                (Supplier<?>) () -> new ParameterizedMessage(
-                                    "[{}] received a remote error as a response to ping {}", sendPingsHandler.id(), finalNodeToSend), e);
-                        } catch (Exception e) {
-                            logger.warn(
-                                (Supplier<?>) () -> new ParameterizedMessage(
-                                    "[{}] failed send ping to {}", sendPingsHandler.id(), finalNodeToSend), e);
-                        } finally {
-                            if (!success) {
-                                latch.countDown();
-                            }
+
+                if (connection == null) {
+                    logger.trace("[{}] connecting (light) to {}", pingingRound.id(), node);
```

Comment:
do we need this trace log here and if so can we fix it to say `temporarily` or something like this